src/pathOrderOptimizer.cpp

```diff
@@ -160,15 +162,23 @@ int PathOrderOptimizer::getRandomPointInPolygon(int poly_idx)
     return rand() % polygons[poly_idx]->size();
 }
 
+static inline bool pointsAreCoincident(const Point& a, const Point& b)
+{
+    return vSize2(a - b) < 25; // points are closer than 5uM, consider them coincident
```

Comment:
I also found that this magic number in `infill/ZigzagConnectorProcessor.h` on line 204 `ZigzagConnectorProcessor::addLine()`. I guess this function can be moved to some common place like `utils/magic.h` so we only have one source of magic, although it is not really the work for this PR.

Reply:
I agree.